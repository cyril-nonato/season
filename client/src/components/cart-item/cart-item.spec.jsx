import React from 'react'
import { fireEvent, render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import CartItem from './cart-item.component'
import 'jest-styled-components'

afterEach(cleanup);

describe('Cart item Test', () => {
  const testItem = {
    index: 1,
    imageUrl: 'some-img.jpg',
    name: 'name',
    price: 200,
    quantity: 2
  }

  let wrapper;
  const functions = {
    addItemToCart: jest.fn(),
    removeItemFromCart: jest.fn()
  }

  beforeEach(() => {
    wrapper = render(<CartItem {...functions} item={testItem} />);
  })

  it('Should renders the props', () => {
    const {getByTestId} = wrapper;
    expect(getByTestId('name')).toHaveTextContent(testItem.name)
    expect(getByTestId('quantity')).toHaveTextContent(testItem.quantity)
  });

  it('Should render the component', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot()
  });

  it('Should call add function', () => {
    const {getByTestId} = wrapper;
    fireEvent.click(getByTestId('addItemToCart'));
    expect(functions.addItemToCart).toHaveBeenCalledTimes(1);
    expect(functions.addItemToCart).toHaveBeenCalledWith(testItem);
  });

  it('Should call remove function', () => {
    const {getByTestId} = wrapper;
    fireEvent.click(getByTestId('removeItemFromCart'));
    expect(functions.removeItemFromCart).toHaveBeenCalledTimes(1);
    expect(functions.removeItemFromCart).toHaveBeenCalledWith(testItem);
  });

});