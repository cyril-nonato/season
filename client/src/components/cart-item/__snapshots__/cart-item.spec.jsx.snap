// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart item Test Should render the component 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  width: 100%;
  grid-template-columns: 1fr 2fr 2fr 1fr 1fr 1fr;
  grid-auto-rows: 15rem;
  font-size: 1.8rem;
}

.c1 {
  justify-self: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c2 {
  height: 100%;
  justify-self: center;
}

.c3 {
  text-transform: uppercase;
  justify-self: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  text-align: center;
}

.c7 {
  justify-self: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4 {
  justify-self: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 1.4rem;
}

.c5:hover {
  cursor: pointer;
}

.c6 {
  margin: 0 .5rem;
}

.c8 {
  justify-self: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c8:hover {
  cursor: pointer;
}

<li
    class="c0"
  >
    <span
      class="c1"
    />
    <img
      alt="name"
      class="c2"
      src="some-img.jpg"
    />
    <span
      class="c3"
      data-testid="name"
    >
      name
    </span>
    <div
      class="c4"
    >
      <span
        class="c5"
        data-testid="removeItemFromCart"
      >
        ❮
      </span>
      <span
        class="c6"
        data-testid="quantity"
      >
        2
      </span>
      <span
        class="c5"
        data-testid="addItemToCart"
      >
        ❯
      </span>
    </div>
    <span
      class="c7"
    >
      200.00
    </span>
    <div
      class="c8"
    >
      ✕
    </div>
  </li>
</DocumentFragment>
`;
